pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -284,25 +290,78 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 
 	// Render the desired objects from the CRD and create or update them.
 	reqLogger.V(3).Info("rendering components")
+	var components []render.Component
 	component, err := render.APIServer(k8sapi.Endpoint, network, false, managementCluster, managementClusterConnection, amazon, tlsSecret, pullSecrets, r.provider == operatorv1.ProviderOpenShift,
 		tunnelCASecret, r.clusterDomain)
 	if err != nil {
 		log.Error(err, "Error rendering APIServer")
 		r.status.SetDegraded("Error rendering APIServer", err.Error())
 		return reconcile.Result{}, err
 	}
+	components = append(components, component)
+
+	if variant == operatorv1.TigeraSecureEnterprise {
+
+		var serverCertSecret *v1.Secret
+		if network.CertificateManagement == nil {
+			serverCertSecret, err = utils.ValidateCertPair(r.client,
+				rmeta.OperatorNamespace(),
+				render.PacketCaptureCertSecret,
+				v1.TLSPrivateKeyKey,
+				v1.TLSCertKey,
+			)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("failed to retrieve / validate %s", render.PacketCaptureCertSecret))
+				r.status.SetDegraded(fmt.Sprintf("failed to retrieve / validate  %s", render.PacketCaptureCertSecret), err.Error())
+				return reconcile.Result{}, err
+			}
+
+			// Create the cert if doesn't exist. If the cert exists, check that the cert
+			// has the expected DNS names. If the cert doesn't and the cert is managed by the
+			// operator, the cert is recreated and returned. If the invalid cert is supplied by
+			// the user, set the component degraded.
+
+			serverCertSecret, err = utils.EnsureCertificateSecret(
+				render.PacketCaptureCertSecret, serverCertSecret, v1.TLSPrivateKeyKey, v1.TLSCertKey, rmeta.DefaultCertificateDuration, dns.GetServiceDNSNames(render.PacketCaptureServiceName, render.PacketCaptureNamespace, r.clusterDomain)...,
+			)
+			if err != nil {
+				r.status.SetDegraded(fmt.Sprintf("Error ensuring packetcapture-api TLS certificate %q exists and has valid DNS names", render.PacketCaptureCertSecret), err.Error())
+				return reconcile.Result{}, err
+			}
+		} else {
+			serverCertSecret = render.CreateCertificateSecret(network.CertificateManagement.CACert, render.PacketCaptureCertSecret, rmeta.OperatorNamespace())
+		}
+
+		// Fetch the Authentication spec. If present, we use to configure user authentication.
+		authenticationCR, err := utils.GetAuthentication(ctx, r.client)
+		if err != nil && !errors.IsNotFound(err) {
+			r.status.SetDegraded("Error querying Authentication", err.Error())
+			return reconcile.Result{}, err
+		}
+
+		keyValidatorConfig, err := utils.GetKeyValidatorConfig(ctx, r.client, authenticationCR, r.clusterDomain)
+		if err != nil {
+			log.Error(err, "Failed to process the authentication CR.")
+			r.status.SetDegraded("Failed to process the authentication CR.", err.Error())
+			return reconcile.Result{}, err
+		}
+
+		var pc = render.PacketCaptureAPI(pullSecrets, r.provider == operatorv1.ProviderOpenShift, network, keyValidatorConfig, serverCertSecret, r.clusterDomain)
```

Comment:
I'm not sure that this should be managed by the apiserver controller. What the reasoning that it should be handled by the apiserver controller and not it's own? 

The biggest red flag for me is that it's not even in the same namespace as the apiserver, so what does it have to do with it?

Reply:
I do not think we want to introduce a new installation CR for this. This service should be part of the default installation of Calico Enterprise.  Creating PacketCaptures is part of the default installation. The service that allows you to download capture files should also be part of the default installation. We will also have a default UI integration which needs this service to be present.